src/cms/templates/pages/page_form.html

```diff
@@ -182,27 +182,24 @@ <h2 class="heading font-normal">
                             </div>
                         {% endif %}
                     {% endif %}
-                    <label for="{{ page_translation_form.slug.id_for_label }}" class="block mb-2 font-bold">
-                        {{ page_translation_form.slug.label }}
-                        <button type="button" class="p-1 bg-gray-200 rounded ml-2"
-                            onclick='document.getElementById("id_slug").readOnly = false;'>
-                            <i data-feather="edit-2" class="inline-block"></i>
+                    <div id="link-container" class="flex items-center">
+                        <label for="{{ page_translation_form.slug.id_for_label }}" class="mr-2 font-bold">
+                            {{ page_translation_form.slug.label }}:
+                        </label>
+                        <div class="mr-2">
+                            {% spaceless %}
+                            <span>{{ page_link }}</span>
+                            {% render_field page_translation_form.slug|add_error_class:"slug-error" class="inline appearance-none rounded" size=page_translation_form.instance.slug|length %}
+                            {% endspaceless %}
+                        </div>
+                        <button id="edit-btn" type="button" class="mr-4">
+                            <i data-feather="edit-3"></i>
+                        </button>
+                        <button id="copy-btn" type="button">
```

Comment:
```suggestion
                        <button id="edit-btn" type="button" class="mr-4 text-gray-800 hover:text-blue-500">
                            <i data-feather="edit-3"></i>
                        </button>
                        <button id="copy-btn" type="button" class="text-gray-800 hover:text-blue-500">
```
I'd also add hover effects...